ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Cleaner.java

```diff
@@ -272,4 +302,22 @@ private void removeFiles(String location, ValidWriteIdList writeIdList, Compacti
       fs.delete(dead, true);
     }
   }
+
+  /**
+   * Check if user configured retention time for the cleanup of obsolete directories/files for the table
+   * has passed or not
+   *
+   * @param ci CompactionInfo
+   * @return True, if retention time has passed and it is ok to clean, else false
+   */
+  public boolean isReadyToCleanWithRetentionPolicy(CompactionInfo ci) {
```

Comment:
This whole thing could be added to the findReadyToClean's sql query where clause. It could use the TxnDbUtil.getEpochFn so there would be no need for new fields in CompactionInfo